qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/template/datamodel/TemplateGlobalAnnotationSupport.java

```diff
@@ -0,0 +1,186 @@
+
+	private static final Logger LOGGER = Logger.getLogger(TemplateGlobalAnnotationSupport.class.getName());
+
+	private static final String[] ANNOTATION_NAMES = {
+		TEMPLATE_GLOBAL_ANNOTATION
+	};
+
+	@Override
+	protected String[] getAnnotationNames() {
+		return ANNOTATION_NAMES;
+	}
+
+	@Override
+	protected void processAnnotation(IJavaElement javaElement, IAnnotation annotation, String annotationName,
+		SearchContext context, IProgressMonitor monitor) throws JavaModelException {
+		if (!(javaElement instanceof IAnnotatable)) {
+			return;
+		}
+		IAnnotation templateGlobal = AnnotationUtils.getAnnotation((IAnnotatable) javaElement,
+			TEMPLATE_GLOBAL_ANNOTATION);
+		if (templateGlobal == null) {
+			return;
+		}
+		if (javaElement instanceof IType) {
+			IType type = (IType) javaElement;
+			collectResolversForTemplateGlobal(type, templateGlobal, context.getDataModelProject().getValueResolvers(),
+				monitor);
+		} else {
+			ITypeResolver typeResolver = QuteSupportForTemplate.createTypeResolver((IType) javaElement);
+
+			if (javaElement instanceof IMethod) {
+				IMethod method = (IMethod) javaElement;
+				collectResolversForTemplateGlobal(method, context.getDataModelProject().getValueResolvers(),
+					typeResolver, monitor);
+			} else if (javaElement instanceof IField) {
+				IField field = (IField) javaElement;
+				collectResolversForTemplateGlobal(field, context.getDataModelProject().getValueResolvers(),
+					typeResolver, monitor);
+			}
+		}
+	}
+
+	private void collectResolversForTemplateGlobal(IType type, IAnnotation templateGlobal,
+		List<ValueResolverInfo> resolvers, IProgressMonitor monitor) {
+		try {
+			ITypeResolver typeResolver = QuteSupportForTemplate.createTypeResolver(type);
+			IField[] fields = type.getFields();
+			for (IField field : fields) {
+				if (isTemplateGlobalMember(field)) {
+					collectResolversForTemplateGlobal(field, resolvers, typeResolver, monitor);
+				}
+			}
+			IMethod[] methods = type.getMethods();
+			for (IMethod method : methods) {
+				if (isTemplateGlobalMember(method)) {
+					collectResolversForTemplateGlobal(method, resolvers, typeResolver, monitor);
+				}
+			}
+		} catch (JavaModelException e) {
+			LOGGER.log(Level.SEVERE, "Error while getting methods of '" + type.getElementName() + "'.", e);
+		}
+	}
+
+	private void collectResolversForTemplateGlobal(IMember member, List<ValueResolverInfo> resolvers,
+		ITypeResolver typeResolver, IProgressMonitor monitor) {
+		// every non-void non-private static method that declares no parameters and
+		// every non-private static field is considered a global variable
+		if (isTemplateGlobalMember(member)) {
+			String sourceType = member.getDeclaringType().getFullyQualifiedName();
+			ValueResolverInfo resolver = new ValueResolverInfo();
+			resolver.setSourceType(sourceType);
+			resolver.setSignature(getSignature(member, typeResolver));
+			// The name is defaulted, i.e. the name of the field/method is used.

```

Comment:
`resolver.setGlobalVariable(true);`

Reply:
It seems that you can override the name of @TemplateGlobal

```
@TemplateGlobal("foo");
private String bar;
```

Your value resolver doesn't take care of that. You should get the "foo" value and if it is not empty, you should set the value to the generated resolver:

```
IAnnotation templateGlobal = AnnotationUtils.getAnnotation(member,TEMPLATE_GLOBAL_ANNOTATION);
if (templateGlobal  != null) {
    // get named from @TemplateGlobal
    String named = ...
    resolver.setNamed(named);
}
```

Reply:
Quarkus doesn't seem to support the declaration of `@TemplateGlobal("foo")`, and the compilation fails when I build the binary:
![Screenshot from 2022-05-26 11-22-02](https://user-images.githubusercontent.com/26389510/170520086-8ab20774-69b6-42be-aeba-3f681648670a.png)
Should this be  "name" instead?

Reply:
Yes sorry, I meant "name" attribute.